packages/xo-server-netbox/src/index.js

```diff
@@ -233,6 +239,107 @@ class Netbox {
 
+    // Tenants -----------------------------------------------------------------
+
+    let nbTenants
+    if (this.#syncUsers) {
+      log.info('Synchronizing users')
+
+      const createNbTenant = xoUser => {
+        const name = xoUser.email.slice(0, NAME_MAX_LENGTH)
+        return {
+          custom_fields: { uuid: xoUser.id },
+          name,
+          slug: slugify(name),
+          description: 'XO user',
+        }
+      }
+
+      const xoUsers = await this.#xo.getAllUsers()
+
+      nbTenants = keyBy(await this.#request('/tenancy/tenants/'), 'custom_fields.uuid')
+      delete nbTenants.null // Ignore tenants that don't have a UUID
+
+      const nbTenantsToCheck = { ...nbTenants }
+
+      const tenantsToUpdate = []
+      const tenantsToCreate = []
+      for (const xoUser of xoUsers) {
+        const nbTenant = nbTenants[xoUser.id]
+        delete nbTenantsToCheck[xoUser.id]
+
+        const updatedTenant = createNbTenant(xoUser)
+
+        if (nbTenant !== undefined) {
+          // Tenant was found in Netbox: update it
+          const patch = diff(updatedTenant, nbTenant)
+          if (patch !== undefined) {
+            tenantsToUpdate.push(patch)
+          }
+        } else {
+          // Tenant wasn't found: create it
+          tenantsToCreate.push(updatedTenant)
+        }
+      }
+
+      // Delete all the other tenants that weren't found in XO
+      const tenantsToDelete = Object.values(nbTenantsToCheck)
+
+      // If a tenant is assigned to a VM (dependentTenants), we must unassign it first.
+      // If a tenant is assigned to another type of object (nonDeletableTenants), we simply log an error.
+      const nonDeletableTenants = []
+      const dependentTenants = []
+      const nonDependentTenants = []
+      for (const nbTenant of tenantsToDelete) {
+        if (
+          (nbTenant.circuit_count ?? 0) +
+            (nbTenant.device_count ?? 0) +
+            (nbTenant.ipaddress_count ?? 0) +
+            (nbTenant.prefix_count ?? 0) +
+            (nbTenant.rack_count ?? 0) +
+            (nbTenant.site_count ?? 0) +
+            (nbTenant.vlan_count ?? 0) +
+            (nbTenant.vrf_count ?? 0) +
+            (nbTenant.cluster_count ?? 0) >
+          0
+        ) {
+          nonDeletableTenants.push(nbTenant)
+        } else if ((nbTenant.virtualmachine_count ?? 0) > 0) {
+          dependentTenants.push(nbTenant)
+        } else {
+          nonDependentTenants.push(nbTenant)
+        }
+      }
+
+      if (nonDeletableTenants.length > 0) {
+        log.warn(`Could not delete ${nonDeletableTenants.length} tenants because dependent object count is not 0`, {
+          tenant: nonDeletableTenants[0],
+        })
+      }
+
+      const nbVms = await this.#request('/virtualization/virtual-machines/')
+
+      const vmsToUpdate = []
+      for (const nbVm of nbVms) {
+        if (find(dependentTenants, { id: nbVm.tenant?.id }) !== undefined) {
```

Comment:
`some`.

And we should avoid loops inside loops as it grows exponentially ideally. Don't change it now but keep this in mind for future devs.